programs/server/Server.cpp

```diff
@@ -1362,6 +1362,7 @@ try
                 cgroups_memory_usage_observer->setLimits(
                     static_cast<uint64_t>(max_server_memory_usage * hard_limit_ratio),
                     static_cast<uint64_t>(max_server_memory_usage * soft_limit_ratio));
+                cgroups_memory_usage_observer->startThread();
```

Comment:
Having to start the thread explicitly seems error-prone (easy to forget). What was the problem with stopping/starting the thread in `setLimits()`?

Reply:
Because I want to reload config when observing memory limit changes --- it is called by `runThread` and already holding the thread mutex.
 
We can write a test to make sure this observer is working.